Formula/borgbackup.rb

```diff
@@ -0,0 +1,32 @@
+class Borgbackup < Formula
+  include Language::Python::Virtualenv
+
+  desc "Deduplicating archiver with compression and authenticated encryption"
+  homepage "https://borgbackup.readthedocs.io/en/stable/"
+  url "https://github.com/borgbackup/borg/releases/download/1.1.13/borgbackup-1.1.13.tar.gz"
+  sha256 "164a8666a61071ce2fa6c60627c7646f12e3a8e74cd38f046be72f5ea91b3821"
+  license "BSD-3-Clause"
+
+  depends_on "pkg-config" => :build
+  depends_on "openssl@1.1"
+  depends_on :osxfuse
+  depends_on "python@3.8"
+
+  resource "msgpack" do
+    url "https://files.pythonhosted.org/packages/74/0a/de673c1c987f5779b65ef69052331ec0b0ebd22958bda77a8284be831964/msgpack-0.6.2.tar.gz"
+    sha256 "ea3c2f859346fcd55fc46e96885301d9c2f7a36d453f5d8f2967840efa1e1830"
+  end
+
+  resource "llfuse" do
+    url "https://files.pythonhosted.org/packages/75/b4/5248459ec0e7e1608814915479cb13e5baf89034b572e3d74d5c9219dd31/llfuse-1.3.6.tar.bz2"
+    sha256 "31a267f7ec542b0cd62e0f1268e1880fdabf3f418ec9447def99acfa6eff2ec9"
+  end
+
+  def install
+    virtualenv_install_with_resources
+  end
+
+  test do
+    system "#{bin}/borg", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
borg has some builtin tests (mostly for the low-level C code), they can be seen at DEBUG log level:
```
$ borg --debug help
using builtin fallback logging configuration
35 self tests completed in 0.11 seconds                  <--------------- !!!
usage: borg [-V] [-h] [--critical] [--error] [--warning] [--info] [--debug]
[...]
```
Note:
- due to the way argparsing/dispatching works, just `borg --version` or `borg --help` does not invoke the self tests.
- these are just very quick / lowlevel tests. a higher level init/create/extract test might be more useful.

Reply:
Yeah, we don't want to run a test suite. We want to see if it works if we do something a user would.

Reply:
Agree. I expanded the test to do a full init, create, extract with one of the test fixtures.